bots/s3_block_public_access.py

```diff
@@ -0,0 +1,122 @@
+
+    Block public access to buckets and objects granted through any existing ACLs and Bucket Policies
+    ( granted through new public ACLs and Bucket Policies Configuration Stays the same ):
+    AUTO: s3_block_public_access
+
+    Block public access to buckets and objects granted through any existing ACLs and Bucket Policies But not for
+    new public ACLs and Bucket Policies:
+    AUTO: s3_block_public_access BlockPublicAcls_NewPUTRequests=false  BlockPublicPolicy_NewPUTRequests=false
+
+"""
+
+import boto3
+from botocore.exceptions import ClientError
+
+usage = 'AUTO: s3_block_public_access BlockPublicAcls_NewPUTRequests=<true|false>  BlockPublicPolicy_NewPUTRequests=<true|false> \n '
+
+text_output = ''
+
+TEXT_OUTPUT_DIC = {
+    'BlockPublicAcls': ' set to true - certain PUT requests includes a public ACL will fail \n ',
+    'IgnorePublicAcls': ' set to true -  S3 will ignore all public ACLs on this bucket and objects in this bucket \n ',
+    'BlockPublicPolicy': 'set to true -  S3 will reject calls to PUT Bucket policy if the specified bucket policy '
+                         'allow public access. \n ',
+    'RestrictPublicBuckets': 'set to true -  access to this bucket restricted to only AWS services and authorized users '
+                             'within this account if the bucket has a public policy. \n '
+}
+
+
+def run_action(boto_session, rule, entity, params):
+    # Create an S3 client
+    s3 = boto3.client('s3')
+    bucket_id = entity['id']
+
+    # For basic findings use  - entity region is not specified
+    region = s3.get_bucket_location(Bucket=bucket_id)['LocationConstraint']
+    s3_client = boto_session.client('s3', region_name=region)
+    current_acces_block_config = s3_client.get_public_access_block(Bucket=bucket_id)['PublicAccessBlockConfiguration']
+    global text_output, usage
+    # Param retrieving
+    try:
+        params_dic = get_params(params, current_acces_block_config)
+    except Exception as e:
+        return (e)
+
+    print(params_dic)
+
+    # -----------------------  Block public access of the bucket ----------------------------------#
+
+    try:
+        # -------call S3 to block public access --------#
+
+        # define changes in acl
+        for key in params_dic:
+            if params_dic[key]:
+                text_output = text_output + key + ' ' + TEXT_OUTPUT_DIC[key]
+
+        result = s3_client.put_public_access_block(
+            Bucket=bucket_id,
+            PublicAccessBlockConfiguration=params_dic
+        )
+
+        responseCode = result['ResponseMetadata']['HTTPStatusCode']
+        if responseCode >= 400:
+            text_output = "Unexpected error: %s \n" % str(result)
+        else:
+            text_output = text_output + "Bucket's Public Access Block enabled: %s \n" % bucket_id
+
+    except ClientError as e:
+        text_output = "Unexpected error: %s \n" % e
+
+    return text_output
+
+
+def get_params(params, curr_config):
```

Comment:
Not straight forward, if you will remove the current convention to  AUTO: s3_block_public_access false false most the function will be redundant and you will just need to unpack the list

Reply:
because I want to allow the user to keep the current configuration in the 2 input parameters in need the function to be more complex and check for either '-' 'false' 'true . also the function checks the input parameters better and returns a better output message. I did change it to be more suitable for the new usage

Reply:
I dont understand why you can set bot as :
AUTO: s3_block_public_access BlockPublicAcls_NewPUTRequests=true  BlockPublicPolicy_NewPUTRequests=true
and also as
AUTO: s3_block_public_access false false

Anyway get_params function seems to be complex and hard to maintain. try to use better naming, also you might want to consider split the bot into few separate easy and readable bots.

Reply:
this is the current usage :
![image](https://user-images.githubusercontent.com/44259622/84402968-cf424300-ac0d-11ea-9f5a-899fc0a3a4fe.png)

there are 4 options in total :
block public access from existing Acls
block public access from existing Bucket policies
block public access from new Acls
block public access from new Bucket policies

so I thought to allow the user to have a choice in blocking the last two(new Acls and policies) 
But the first 2 will be always true that the bucket will be blocked for current public access.
if you suggest a different approach I will change it. 
(maybe get a list of 4 params, and ill just unpack it and pass it to AWS ) 
Anyway, I will remove the get param function (although I took the Idea from an existing bo)t. I will only unpack the list.

Reply:
I don't see it in the bot's documentation.
About the bot configuration you should advise Guy because he understand AWS and the users need more than I do, but I think the bot should be as simple as possible both from the configurations side and from code side.

Reply:
following to consulting with Guy, He suggested that we can create 2 bots: one basic and one more complex with more detailed documentation, I think ill start with creating a separate pull request for the basic bot and we will proceed from there